docs/topics/pagination.txt

```diff
@@ -81,72 +81,71 @@ accessing the items for each page::
 your view class, for example::
 
     from django.views.generic import ListView
 
-    from myapp.models import Contacts
+    from myapp.models import Contact
 
-    class ContactsList(ListView):
+    class ContactList(ListView):
         paginate_by = 2
-        model = Contacts
+        model = Contact
 
-The only thing your users will be missing is a way to navigate to the next or
-previous page. To achieve this, add links to the next and previous page, like
-shown in the below example ``list.html``.
-
-.. _using-paginator-in-view:
-
-Using ``Paginator`` in a view
-=============================
-
-Here's a slightly more complex example using
-:class:`~django.core.paginator.Paginator` in a view to paginate a queryset. We
-give both the view and the accompanying template to show how you can display
-the results. This example assumes you have a ``Contacts`` model that has
-already been imported.
-
-The view function looks like this::
-
-    from django.core.paginator import Paginator
-    from django.shortcuts import render
-
-    def listing(request):
-        contact_list = Contacts.objects.all()
-        paginator = Paginator(contact_list, 25) # Show 25 contacts per page
-
-        page = request.GET.get('page')
-        contacts = paginator.get_page(page)
-        return render(request, 'list.html', {'contacts': contacts})
-
-In the template :file:`list.html`, you'll want to include navigation between
-pages along with any interesting information from the objects themselves:
+This limits the number of objects per page and adds ``'paginator'`` and
+``'page'``  to the ``context``. To allow your users to navigate between pages,
+add links to the next and previous pages, in your template like this:
 
 .. code-block:: html+django
 
-    {% for contact in contacts %}
+    {% for contact in page_obj %}
         {# Each "contact" is a Contact model object. #}
         {{ contact.full_name|upper }}<br>
         ...
     {% endfor %}
 
     <div class="pagination">
         <span class="step-links">
-            {% if contacts.has_previous %}
+            {% if page_obj.has_previous %}
                 <a href="?page=1">&laquo; first</a>
-                <a href="?page={{ contacts.previous_page_number }}">previous</a>
+                <a href="?page={{ page_obj.previous_page_number }}">previous</a>
             {% endif %}
 
             <span class="current">
-                Page {{ contacts.number }} of {{ contacts.paginator.num_pages }}.
+                Page {{ page_obj.number }} of {{ paginator.num_pages }}.
             </span>
 
-            {% if contacts.has_next %}
-                <a href="?page={{ contacts.next_page_number }}">next</a>
-                <a href="?page={{ contacts.paginator.num_pages }}">last &raquo;</a>
+            {% if page_obj.has_next %}
+                <a href="?page={{ page_obj.next_page_number }}">next</a>
+                <a href="?page={{ paginator.num_pages }}">last &raquo;</a>
```

Comment:
I think it's better to keep an example shorter and use `page_obj.paginator` instead of `paginator`, with that we can remove adding `paginator` to the context in the `listing()`.